### PR TITLE
[chore] move tcplog receiver to generated lifecycle tests

### PR DESCRIPTION
**Pull request by atoulme**
Related to https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/27850

### CODE REVIEW

File: receiver/tcplogreceiver/metadata.yaml [hunk 1]
```diff
@@ -7,3 +7,7 @@ status:
   distributions: [contrib, splunk, observiq, sumo]
   codeowners:
     active: [djaglowski]
+
+tests:
+  config:
+    listen_address: "0.0.0.0:0"
```
Comment (bogdandrutu) [inline receiver/tcplogreceiver/metadata.yaml:13]: New line.

**Review by bogdandrutu — APPROVED** [review]

File: receiver/tcplogreceiver/metadata.yaml (hunk 1 above)
Comment (atoulme) [reply receiver/tcplogreceiver/metadata.yaml]: Thanks! added one

Comment (bogdandrutu) [inline receiver/tcplogreceiver/metadata.yaml:13]: Can it be "localhost" instead?

**Review by bogdandrutu — APPROVED** [review]

File: receiver/tcplogreceiver/metadata.yaml (hunk 1 above)
Comment (atoulme) [reply receiver/tcplogreceiver/metadata.yaml]: changed to localhost, good point

**Review by bogdandrutu — APPROVED** [review]

**Pull request merged**